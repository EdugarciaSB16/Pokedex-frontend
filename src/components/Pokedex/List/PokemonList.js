/**
 * Página que muestra todos los pokémons
 *
 */
// Dependencias
import React, { useEffect, useState } from 'react';
// Endpoints
import { getPokemons } from '../../../service/Endpoints';
import { PokemonItem } from '../Item/PokemonItem';
// Componentes
import { Spinner } from '../../Spinner/Spinner';
/**
 * Contenedor de la lista de pokémons
 *
 */
export const PokemonList = () => {
    // Estado donde se almacena el listado de pokémons
    const [pokemons, setPokemons] = useState({});
    // Estado donde se almecena el estado de carga
    const [loading, setLoading] = useState(true);

    // Efecto que realiza la peticion y obtiene el listado de componentes
    useEffect(() => {
        getPokemons().then((pokemon) => {
            setPokemons(pokemon);
            setLoading(false);
        });
    }, []);

    return (
        <div className="PokemonList">
            <header className="List__header">
                <div className="container">
                    <h1>Pokédex</h1>
                </div>
            </header>
            <main className="container List__main">
                {loading ? (
                    <Spinner />
                ) : (
                    <div className="container mt-5 animate__animated animate__fadeIn">
                        <div className="row g-3 row-cols-1 row-cols-sm-2 row-cols-md-3">
                            {pokemons.map((pokemon) => (
                                <div className="col List__card-content" key={pokemon.number}>
                                    <PokemonItem {...pokemon} />
                                </div>
                            ))}
                        </div>
                    </div>
                )}
            </main>
        </div>
    );
};
