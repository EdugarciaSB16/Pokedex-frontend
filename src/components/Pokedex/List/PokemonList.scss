.List__header{
    width: 100%;
    height: 25vh;
    background-image: url('https://cdn.pixabay.com/photo/2016/08/17/20/12/pokemon-1601385_960_720.jpg');
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: center;
    align-items: flex-start;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    color: white;
}

.List__card-content {
    margin: auto;
    width: 300px;
    padding:20px;
}
