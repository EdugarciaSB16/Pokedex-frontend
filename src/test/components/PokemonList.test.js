/**
 * Test al componente PokemonList
 *
 */
// Dependencias
import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
// Componentes
import { PokemonList } from '../../components/Pokedex/List/PokemonList';

configure({ adapter: new Adapter() });

describe('Pruebas en el componente PokemonList', () => {
    const wrapper = shallow(<PokemonList />);

    test('Debe de renderizar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
