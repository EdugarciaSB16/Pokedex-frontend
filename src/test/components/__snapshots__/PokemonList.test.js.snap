// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PokemonList Debe de renderizar el componente correctamente 1`] = `
<div
  className="PokemonList"
>
  <header
    className="List__header"
  >
    <div
      className="container"
    >
      <h1>
        Pok√©dex
      </h1>
    </div>
  </header>
  <main
    className="container List__main"
  >
    <Spinner />
  </main>
</div>
`;
